Add unit-tests to ensure that all the entries in contributors.json are valid

aj062 opened the pull request:
#### 23e738c257d34e7da5a3f0d0d173b42e6e0bd699
<pre>
Add unit-tests to ensure that all the entries in contributors.json are valid
<a href="https://bugs.webkit.org/show_bug.cgi?id=247137">https://bugs.webkit.org/show_bug.cgi?id=247137</a>

Reviewed by Ryan Haddad.

* Tools/Scripts/webkitpy/common/config/contributors_json_unittest.py: Added.
(ContributorsJsonTest.test_valid_contributors_json):
* Tools/CISupport/ews-build/steps.py: Trigger webkitpy ews for contributors.json changes.

Canonical link: <a href="https://commits.webkit.org/256416@main">https://commits.webkit.org/256416@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3b0a9592b6fbdec53c055becaa086e6d465483ef

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95681 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/4941 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/28728 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105260 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/165564 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/99666 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/5014 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/33697 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88059 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/101108 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/101342 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/3684 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82301 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/30737 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/99266 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/85548 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/87457 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/73568 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39431 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/18997 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/94651 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37128 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20310 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/41124 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/42959 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2126 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/43111 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/39561 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

JonWBedard reviewed (commented):
Pretty sure the style-checker does some of this too, can (and probably should) include those checks here.

lauromoura commented on Tools/Scripts/webkitpy/common/config/contributors_json_unittest.py:
```diff
@@ -0,0 +1,49 @@
+class ContributorsJsonTest(unittest.TestCase):
+    def load_json(self):
+        json_path = os.path.join(
+            os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__)))))),
```
```suggestion
            WebKitFinder(FileSystem()).webkit_base()
```
Will keep the test working if it ever changes to a different level in the tree.

aj062 replied:
Done

lauromoura commented on Tools/Scripts/webkitpy/common/config/contributors_json_unittest.py:
```diff
@@ -0,0 +1,49 @@
+import os
+import unittest
+from webkitpy.common.system.filesystem import FileSystem
+
```
```suggestion
from webkitpy.common.webkit_finder import WebKitFinder

```

aj062 replied:
Done

lauromoura reviewed (commented):
Looks good. Minor comment on the `dirname` calls.

aj062 committed:
Add unit-tests to ensure that all the entries in contributors.json are valid
https://bugs.webkit.org/show_bug.cgi?id=247137

Reviewed by Ryan Haddad.

* Tools/Scripts/webkitpy/common/config/contributors_json_unittest.py: Added.
(ContributorsJsonTest.test_valid_contributors_json):
* Tools/CISupport/ews-build/steps.py: Trigger webkitpy ews for contributors.json changes.

Canonical link: https://commits.webkit.org/256416@main
